File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/java_solutions/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int n = arr.length;
        if (n == 0) {
            return arr;
        }

        int maxSoFar = -1;

        for (int i = n - 1; i >= 0; i--) {
            int currentElement = arr[i];
            arr[i] = maxSoFar;
            if (currentElement > maxSoFar) {
                maxSoFar = currentElement;
            }
        }

        return arr;
    }
}

File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/python3_solutions/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: list[int]) -> list[int]:
        max_so_far = -1

        for i in range(len(arr) - 1, -1, -1):
            current_element = arr[i]
            arr[i] = max_so_far
            max_so_far = max(max_so_far, current_element)

        return arr